Multisig/UI/App/UIKit/TransactionlListViewController/TransactionListViewController.swift

```diff
@@ -0,0 +1,206 @@
+    private var loadNextPageDataTask: URLSessionTask?
+
+    private var model = TransactionsListViewModel()
+
+    override var isEmpty: Bool {
+        model.isEmpty
+    }
+
+    convenience init() {
+        self.init(namedClass: Self.superclass())
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        tableView.delegate = self
+        tableView.dataSource = self
+
+        tableView.backgroundColor = .gnoWhite
+
+        tableView.registerCell(TransactionListTableViewCell.self)
+        tableView.registerHeaderFooterView(BasicHeaderView.self)
+
+        tableView.sectionHeaderHeight = BasicHeaderView.headerHeight
+
+        emptyView.setText("Transactions will appear here")
+        emptyView.setImage(#imageLiteral(resourceName: "ico-no-transactions"))
+    }
+
+    override func reloadData() {
+        super.reloadData()
+        loadFirstPageDataTask?.cancel()
+        loadNextPageDataTask?.cancel()
+        stopNextPageLoadingAnimation()
+
+        do {
+            let address = try Address(from: try Safe.getSelected()!.address!)
+
+            loadFirstPageDataTask = clientGatewayService.asyncTransactionList(address: address) { [weak self] result in
+
+                guard let `self` = self else { return }
+                switch result {
+                case .failure(let error):
+                    DispatchQueue.main.async { [weak self] in
+                        guard let `self` = self else { return }
+                        // ignore cancellation error due to cancelling the
+                        // currently running task. Otherwise user will see
+                        // meaningless message.
+                        if (error as NSError).code == URLError.cancelled.rawValue &&
+                            (error as NSError).domain == NSURLErrorDomain {
+                            return
+                        }
+                        self.onError(error)
+                    }
+                case .success(let page):
+                    var model = TransactionsListViewModel(page.results.flatMap { TransactionViewModel.create(from: $0) })
+                    model.next = page.next
+
+                    DispatchQueue.main.async { [weak self] in
+                        guard let `self` = self else { return }
+                        self.model = model
+                        self.onSuccess()
+                    }
+                }
+            }
+        } catch {
+            onError(error)
+        }
+    }
+
+    private func startNextPageLoadingAnimation() {
+        let indicator = UIActivityIndicatorView(style: .medium)
+        indicator.startAnimating()
+        indicator.frame = CGRect(origin: .zero, size: CGSize(width: tableView.bounds.width, height: 100))
+        // moves the indicator up
+        indicator.bounds = CGRect(origin: CGPoint(x: 0, y: 30), size: indicator.frame.size)
+        tableView.tableFooterView = indicator
```

Comment:
interesting approach
A possible issue with a footer is that if loading takes some time and you scroll up, then this loading indicator will be sticky to the bottom (because of the grouped style of the table view). But I am not sure if this is an issue.

Reply:
I haven't seen any problems with this so far